roles/maas_postgres/tasks/o11y_pg.yaml

```diff
@@ -0,0 +1,60 @@
+---
+- name: "Create postgres-exporter directory"
+  ansible.builtin.file:
+    path: "{{ o11y_postgres_exporter_dir }}"
+    state: "{{ 'directory' if (o11y_enable and 'maas_postgres' in group_names) else 'absent' }}"
+    mode: 0755
+  changed_when: false
+
+- name: "Discover host architecture"
+  ansible.builtin.command: dpkg --print-architecture
+  register: ar_out
+  changed_when: false
+
+- name: "Set arch facts"
+  ansible.builtin.set_fact:
+    arch: "{{ ar_out.stdout }}"
+    pg_exp_pkg: "https://github.com/prometheus-community/postgres_exporter/releases/download/v0.11.1/postgres_exporter-0.11.1.linux-{{ ar_out.stdout }}.tar.gz"
+
+- name: "Download and unzip latest postgres-exporter release"
+  ansible.builtin.unarchive:
+    src: "{{ pg_exp_pkg }}"
+    dest: "{{ o11y_postgres_exporter_dir }}"
+    remote_src: true
+
+- name: "Set env variable for data source"
+  ansible.builtin.copy:
+    dest: "{{ o11y_postgres_exporter_dir }}/postgres_exporter-0.11.1.linux-{{arch}}/postgres_exporter.env"
+    content: DATA_SOURCE_NAME="postgresql://{{ maas_postgres_user }}:{{ maas_postgres_password }}@{{ inventory_hostname }}:5432/?sslmode=disable"
+
+- name: "Create a service file"
+  ansible.builtin.copy:
+    dest: /etc/systemd/system/pg-exporter.service
+    mode: 0644
+    content: |
+      [Unit]
+      Description=postgres-exporter service
+      Requires=network.target
```

Comment:
we need to check the postgres service unit name, and use it here. This way the agent waits for the DB to start